### PR TITLE
Regenerate and reindex reference numbers on private dossiers

### LINKED ISSUE #3897: Upgrade existing content in private folder.
The reference number for private content was changed in https://github.com/4teamwork/opengever.core/pull/3883. However existing content is not reindexed. We should also reindex and update the reference number for the current content as the reference will change.

### PR DESCRIPTION
**Pull request by Rotonen**
Nicked the logic from: https://github.com/4teamwork/opengever.core/blob/a3eb6bb6b41d710441664de50bcd003c6f8ae1f7/opengever/dossier/handlers.py#L57-L79

Closes #3897

### CODE REVIEW

**Comment by Rotonen** [comment]
⚠️  Do not merge, I'm on a hunt to figure out why the private folder does not actually use the reference number adapter I'm expecting it to use. ⚠️

**Comment by Rotonen** [comment]
@deiferni ready for review.

**Comment by deiferni** [comment]
also tested locally, seems to work as expected 👍

File: opengever/core/upgrades/20180129162603_regenerate_and_reindex_reference_numbers_on_private_roots_and_their_contents/upgrade.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from Acquisition import aq_parent
+from ftw.upgrade import UpgradeStep
+from opengever.base.interfaces import IReferenceNumberPrefix
+
+
+def regenerate_and_reindex_reference_number(obj):
+    parent = aq_parent(aq_inner(obj))
+    prefix_adapter = IReferenceNumberPrefix(parent)
+
+    prefix_adapter.set_number(obj)
```
[the remark this answers is not in the document]
Comment (Rotonen) [reply opengever/core/upgrades/20180129162603_regenerate_and_reindex_reference_numbers_on_private_roots_and_their_contents/upgrade.py]: The logic is purposefully nicked from the dossier reference number prefix handling linked to in the pull request body as that path ends up reusing where ever possible.

**Comment by Rotonen** [comment]
@deiferni ready for rereview.

**Pull request merged**